src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/Orchestrator/DateTimeAnalyzer.cs

```diff
@@ -20,43 +20,36 @@ public class DateTimeAnalyzer : DiagnosticAnalyzer
         private const string Category = SupportedCategories.Orchestrator;
         public const DiagnosticSeverity severity = DiagnosticSeverity.Warning;
 
-        private static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, severity, isEnabledByDefault: true, description: Description);
+        public static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, severity, isEnabledByDefault: true, description: Description);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
-
-        public override void Initialize(AnalysisContext context)
+        public static bool RegisterDiagnostic(SyntaxNode method, CompilationAnalysisContext context, SemanticModel semanticModel)
         {
-            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
-            context.EnableConcurrentExecution();
-            context.RegisterSyntaxNodeAction(AnalyzeIdentifierDateTimeNow, SyntaxKind.IdentifierName);
-        }
+            var diagnosedIssue = false;
 
-        private static void AnalyzeIdentifierDateTimeNow(SyntaxNodeAnalysisContext context)
-        {
-            var identifierName = context.Node as IdentifierNameSyntax;
-            if (identifierName != null)
+            foreach (SyntaxNode descendant in method.DescendantNodes())
             {
-                var identifierText = identifierName.Identifier.ValueText;
-                if (identifierText == "Now" || identifierText == "UtcNow" || identifierText == "Today")
+                var identifierName = descendant as IdentifierNameSyntax;
+                if (identifierName != null)
                 {
-                    var memberAccessExpression = identifierName.Parent;
-                    var memberSymbol = context.SemanticModel.GetSymbolInfo(memberAccessExpression).Symbol;
-
-                    if (memberSymbol == null || !memberSymbol.ToString().StartsWith("System.DateTime"))
+                    var identifierText = identifierName.Identifier.ValueText;
+                    if (identifierText == "Now" || identifierText == "UtcNow" || identifierText == "Today")
                     {
-                        return;
-                    }
+                        var memberAccessExpression = identifierName.Parent;
+                        var memberSymbol = semanticModel.GetSymbolInfo(memberAccessExpression).Symbol;
 
-                    if (!SyntaxNodeUtils.IsInsideOrchestrator(identifierName) && !SyntaxNodeUtils.IsMarkedDeterministic(identifierName))
-                    {
-                        return;
-                    }
+                        if (memberSymbol != null && memberSymbol.ToString().StartsWith("System.DateTime"))
```

Comment:
Are you using `.StartsWith("System.DateTime")` to account for both `System.DateTime` and `System.DateTimeOffset`? If so, I recommend adding a comment to make it clear that you're doing this intentionally.